### PR TITLE
Timestamp parsing

### PR DESCRIPTION
**Pull request by foram-splunk**

### CODE REVIEW

File: src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java [hunk 1]
```diff
@@ -497,4 +521,28 @@ private void validateTopicsAndTopicsRegexCombination(Map<String, String> taskCon
...
 
-}
+    private void validateRegexForTimestamp(String regex){
+            if(enableTimestampExtraction &&  regex == null ){
+                throw new ConfigException("For Timestamp Extraction regex value should not be null");
+            }
+            
+            if(StringUtils.isNotBlank(regex) && enableTimestampExtraction){
+                if(Pattern.compile(regex) == null){
+                    throw new ConfigException(" Invalid Regex please enter again");
```
Comment (artemrys) [inline src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java:531]: nit: extra space in the beginning?

File: src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java [hunk 2]
```diff
@@ -497,4 +521,28 @@ private void validateTopicsAndTopicsRegexCombination(Map<String, String> taskCon
...
+
+    private static boolean getNamedGroupCandidates(String regex) {
+        Matcher m = Pattern.compile("\\(\\?<([a-zA-Z][a-zA-Z0-9]*)>").matcher(regex);
+            while (m.find()) {
+                if(m.group(1).equals("time")) {
+                    return true;   
+                }             
+            }
+            return false;
+        }
```
Comment (artemrys) [inline src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java:547]: This one seems to be formatted strangely.

Comment (artemrys) [reply src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java:547]: Does Java have some code formatter that we can use to format all the code?

File: test/testcases/test_data_onboarding.py [hunk 3]
```diff
@@ -44,3 +44,38 @@ def test_proto_data_onboarding(self, setup, test_scenario, test_input, expected)
                                           password=setup["splunk_password"])
         logger.info("Splunk received %s events in the last hour", len(events))
         assert len(events) == expected
+
+    @pytest.mark.parametrize("test_scenario, test_input, expected", [
```
Comment (artemrys) [inline test/testcases/test_data_onboarding.py:48]: What do you think about not using `pytest.mark.parametrize` if you have just 1 thing to parametrize? Here and below.

File: test/testcases/test_data_onboarding.py [hunk 4]
```diff
@@ -44,3 +44,38 @@ def test_proto_data_onboarding(self, setup, test_scenario, test_input, expected)
...
+        ("date_format", "latest=1365209605.000", "2010-06-13T23:11:52.454+00:00"),
+    ])
+    def test_extracted_timestamp_data_onboarding_date_format(self, setup, test_scenario, test_input, expected):
+        logger.info("testing {0} input={1} expected={2} event(s)".format(test_scenario, test_input, expected))
```
Comment (artemrys) [inline test/testcases/test_data_onboarding.py:52]: If we are Python 3.7+, we can use f-strings, here and everywhere in the code when it is applicable.

File: src/test/java/com/splunk/kafka/connect/SplunkSinkTaskTest.java [hunk 5]
```diff
@@ -246,6 +249,91 @@ public void putWithRawAndAck() {
...
+        task.start(config);
+        task.put(record);
+       
+        List<EventBatch> batches = hec.getBatches();
+        for (Iterator<EventBatch> iter = batches.listIterator(); iter.hasNext();) {
+            EventBatch ch = iter.next();
+            List<Event> ev = ch.getEvents();
+            Iterator<Event> iter1 = ev.listIterator() ;
+            Event ev1 = iter1.next();
+            Assert.assertEquals(1.276470712454E12+"",ev1.getTime()*1000+"");
```
Comment (artemrys) [inline src/test/java/com/splunk/kafka/connect/SplunkSinkTaskTest.java:275]: Don't have much experience with Java, but why do we need to `+""` here to compare values?

**Review by artemrys — CHANGES_REQUESTED** [review]

**Comment by artemrys** [comment]
@foram-splunk Can you please add a comment to my comments which you have resolved?

File: test/testcases/test_data_onboarding.py (hunk 4 above)
Comment (foram-splunk) [reply test/testcases/test_data_onboarding.py]: Done. Updated the code using f-strings.Thanks.

File: src/test/java/com/splunk/kafka/connect/SplunkSinkTaskTest.java (hunk 5 above)
Comment (foram-splunk) [reply src/test/java/com/splunk/kafka/connect/SplunkSinkTaskTest.java]: Resolved it here: 
https://github.com/splunk/kafka-connect-splunk/blob/timestamp-parsing/src/test/java/com/splunk/kafka/connect/SplunkSinkTaskTest.java#L272

("" was to convert it to string.Used double precision instead)

File: src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java (hunk 2 above)
Comment (foram-splunk) [reply src/main/java/com/splunk/kafka/connect/SplunkSinkConnectorConfig.java]: Updated formatting.

**Comment by foram-splunk** [comment]
> @foram-splunk Can you please add a comment to my comments which you have resolved?

Done @artemrys

**Review by artemrys — APPROVED** [review]

**Pull request merged**